workflow/executor/agent.go

```diff
@@ -269,7 +269,7 @@ func (ae *AgentExecutor) executeHTTPTemplate(ctx context.Context, tmpl wfv1.Temp
 			"request": map[string]interface{}{
 				"method":  tmpl.HTTP.Method,
 				"url":     tmpl.HTTP.URL,
-				"body":    tmpl.HTTP.Body,
+				"body":    tmpl.HTTP.GetBody(),
```

Comment:
I think we should have a separate `bodyBytes` field.

Reply:
I've added a separate `bodyFrom` field so that we can add support for configmaps, artifacts, secrets in the future.